### PR TITLE
Using a private repository for user containers - MPI File Utils

### PR DESCRIPTION
**Pull request by roehrich-hpe**
Show the placement of the imagePullSecrets lists for private MPI container images in the NnfContainerProfile resource.

### CODE REVIEW

File: docs/guides/user-containers/readme.md [hunk 1]
```diff
@@ -137,3 +137,63 @@ Now any user can select this profile in their Workflow by specifying it in a
...
+          spec:
+            imagePullSecrets:
+            - name: readonly-red-rock-slushy
+            containers:
+            - command:
+              - mpirun
+              - dcmp
+              - $(DW_JOB_foo_local_storage)/0
+              - $(DW_JOB_foo_local_storage)/1
+              image: deanhpe/red-rock-slushy:v2.0
```
[the remark this answers is not in the document]
Comment (roehrich-hpe) [reply docs/guides/user-containers/readme.md:169]: Yep, it was supposed to be "dean", like in the earlier example.

[the remark this answers is not in the document]
Comment (roehrich-hpe) [reply docs/guides/user-containers/readme.md:169]: hmm, this PR isn't picking up new commits to the PR branch...

**Review by ajfloeder — APPROVED** [review]

File: docs/guides/user-containers/readme.md [hunk 2]
```diff
@@ -137,3 +137,63 @@ Now any user can select this profile in their Workflow by specifying it in a
 ```bash
 #DW container profile=red-rock-slushy  [...]
 ```
+
+### Using a Private Version of MPI File Utils
+
+If our user's containerized application instead contains `mpirun` and `dcp`,
+because perhaps it's a private copy of [nnf-mfu](https://github.com/NearNodeFlash/nnf-mfu),
+then the administrator would insert two `imagePullSecrets` lists into the
+`mpiSpec` of the NnfContainerProfile for the MPI launcher and the MPI worker.
```
Comment (bdevcich-hpe) [inline docs/guides/user-containers/readme.md:146]: It's not really about running a customized version of mpirun/dcp, but rather SCR (or any app really). I envision the private repo to house a containerized version of SCR and that SCR image would be based on nnf-mfu.

The [container example hello world](https://github.com/NearNodeFlash/nnf-container-example/blob/master/src/mpi_hello_world.c) isn't SCR, but gives you an idea of what that custom image might contain.

File: docs/guides/user-containers/readme.md [hunk 3]
```diff
@@ -137,3 +137,63 @@ Now any user can select this profile in their Workflow by specifying it in a
 ```bash
 #DW container profile=red-rock-slushy  [...]
 ```
+
+### Using a Private Version of MPI File Utils
```
Comment (bdevcich-hpe) [inline docs/guides/user-containers/readme.md:141]: I would title this Using Private Image Repositories. See my comment below.

**Comment by bdevcich-hpe** [comment]
Thanks for adding this!

File: docs/guides/user-containers/readme.md (hunk 3 above)
Comment (roehrich-hpe) [reply docs/guides/user-containers/readme.md:141]: No, that heading is used above.  This new section is specifically about the adding the imagePullSecrets lists to the mpiSpec in the NnfContainerProfile.

File: docs/guides/user-containers/readme.md (hunk 2 above)
Comment (roehrich-hpe) [reply docs/guides/user-containers/readme.md:146]: This section is specifically about the mpiSpec.

File: docs/guides/user-containers/readme.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (roehrich-hpe) [reply docs/guides/user-containers/readme.md]: Oh, look. the new commits are finally visible in this PR...that's the first time I've seen it have that kind of delay.

File: docs/guides/user-containers/readme.md (hunk 3 above)
Comment (bdevcich-hpe) [reply docs/guides/user-containers/readme.md:141]: Whoops. I would then just title these sections as Adding Secrets for Non-MPI/MPI Containers or something to that affect.

File: docs/guides/user-containers/readme.md (hunk 2 above)
Comment (bdevcich-hpe) [reply docs/guides/user-containers/readme.md:146]: Sure, but I don't understand why were are talking about dcp here.

Comment (roehrich-hpe) [reply docs/guides/user-containers/readme.md:146]: I'll refer to "MPI File Utils" rather than "mpirun" or "dcp"..

**Comment by roehrich-hpe** [comment]
Wow, today there is a long delay before new commits appear in the PR.

**Comment by roehrich-hpe** [comment]
Okay, the PR shows all of my latest updates.

**Review by bdevcich-hpe — APPROVED** [review]

**Pull request merged**